src/set_relation/set_relation_test.cc

```diff
@@ -4410,4 +4411,40 @@ TEST_F(SetRelationTest, getZ3form){
 
    delete s1;
 }
-
+// Test restrict operations.
+
+TEST_F(SetRelationTest, RestrictDomainTest){
+    Relation *rel = new Relation("{[l] -> [k]: 1 <= n and n <= 10 and k = n + 1}");
+    Set * set = new Set("{[i]: 5 <= i and i <= 25}");
+    Relation * restrictRel = rel->Restrict(set);
+
+    EXPECT_EQ( "{ [l] -> [k] : __tv1 - n - 1 = 0"
+               " && -__tv0 + 25 >= 0 && __tv0 - 5 >= 0"
+               " && -n + 10 >= 0 && n - 1 >= 0 }",
+               restrictRel->toString());
+
+     auto set1 = new Set("{[i,j]: i >= 0 and i < NR and"
+                          " j >= 0 and j < NC and Ad(i,j) > 0}");
+     auto rel1 = new Relation("{[i,j] -> [n]:"
+                                    " row(n) = i and col(n) = j and  i >= 0 and "
+                                    " i < NR and j >= 0 and j < NC}");
+     Relation* restrictRel2 = rel1->Restrict(set1);
+     EXPECT_EQ("{ [i, j] -> [n] : i - row(n) = 0 && j - col(n) = 0"
+               " && i >= 0 && j >= 0 && Ad(i, j) - 1 >= 0 && -i + NR"
+               " - 1 >= 0 && -j + NC - 1 >= 0 }",restrictRel2->
+               prettyPrintString());
+
+
+     Relation *rel2 = new Relation("{[l] -> [k]: 1 <= i and i <= 10 and k = i + 1}");
+     Set * set2 = new Set("{[i,l]: 5 <= i and i <= 25 and l = row(i)}");
+
+     // expect exception
+     EXPECT_ANY_THROW(rel2->Restrict(set2));
```

Comment:
Why are we expecting a throw?  What is the output supposed to be?

Reply:
The input arity of the relation and the set's arity must be equal. I will change the test case to actually test for the exception.